File: src/bot/telegram.js
/**
 * telegram.js
 * ------------
 * Envía mensajes a Telegram usando https nativo de Node.js.
 * Sin dependencias externas.
 */

const https = require('https');

async function sendAlert(item) {
  const token  = process.env.TELEGRAM_TOKEN;
  const chatId = process.env.TELEGRAM_CHAT_ID;

  if (!token || !chatId) {
    console.error('[Telegram] TELEGRAM_TOKEN o TELEGRAM_CHAT_ID no definidos.');
    return;
  }

  const platform = item.platform || 'Vestiaire';
  const emoji     = platform === 'Wallapop' ? '\uD83D\uDFE2' : '\uD83D\uDD25';
  const filterLine = item.filterName ? `Filtro: ${item.filterName}\n` : '';

  const text = [
    emoji + ' Nuevo artículo en ' + platform,
    '',
    filterLine + 'Nombre: ' + (item.name || 'Sin nombre'),
    'Precio: ' + (item.price || 'N/D'),
    '',
    item.link || '',
  ].join('\n');

  const body = JSON.stringify({ chat_id: chatId, text });

  return new Promise((resolve) => {
    const req = https.request(
      {
        hostname: 'api.telegram.org',
        path: '/bot' + token + '/sendMessage',
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Content-Length': Buffer.byteLength(body),
        },
      },
      (res) => {
        let data = '';
        res.on('data', (chunk) => { data += chunk; });
        res.on('end', () => {
          if (res.statusCode !== 200) {
            console.error('[Telegram] Error HTTP ' + res.statusCode + ':', data);
          } else {
            console.log('[Telegram] Alerta enviada: ' + item.name);
          }
          resolve();
        });
      }
    );
    req.on('error', (err) => {
      console.error('[Telegram] Error de red:', err.message);
      resolve();
    });
    req.write(body);
    req.end();
  });
}

module.exports = { sendAlert };

File: app/services/telegram.py
"""
telegram.py
-----------
Telegram notification service.
"""

import logging
import os

import requests

logger = logging.getLogger(__name__)

_TELEGRAM_API = "https://api.telegram.org/bot{token}/sendMessage"


def send(product: dict, item: dict) -> None:
    token = os.getenv("TELEGRAM_TOKEN", "").strip()
    chat_id = os.getenv("TELEGRAM_CHAT_ID", "").strip()

    if not token or not chat_id:
        logger.warning("[Telegram] TELEGRAM_TOKEN o TELEGRAM_CHAT_ID no definidos.")
        return

    filter_name = product.get("name", "?")
    marketplace = item.get("marketplace", product.get("marketplace", "wallapop"))
    title = item.get("title", item.get("name", "Sin nombre"))
    url = item.get("url", item.get("link", ""))

    price = item.get("price", "N/D")
    if isinstance(price, (int, float)):
        price_text = f"{price:.2f} EUR"
    else:
        price_text = str(price)

    text = (
        f"Nuevo articulo en {marketplace}\n\n"
        f"Filtro: {filter_name}\n"
        f"Nombre: {title}\n"
        f"Precio: {price_text}\n\n"
        f"{url}"
    )

    try:
        url = _TELEGRAM_API.format(token=token)
        resp = requests.post(
            url,
            json={"chat_id": chat_id, "text": text, "disable_web_page_preview": False},
            timeout=10,
        )
        if resp.status_code == 200:
            logger.info(f"[Telegram] Alerta enviada: {title}")
        else:
            logger.error(f"[Telegram] Error HTTP {resp.status_code}: {resp.text[:200]}")
    except Exception as e:
        logger.error(f"[Telegram] Error enviando mensaje: {e}")
